cmd/idp-ldap-accesskey-create.go

```diff
@@ -128,77 +132,80 @@ func mainIDPLdapAccesskeyCreate(ctx *cli.Context) error {
-	accessVal := ctx.String("access-key")
-	secretVal := ctx.String("secret-key")
 	name := ctx.String("name")
-	description := ctx.String("description")
+	expVal := ctx.String("expiry")
 	policyPath := ctx.String("policy")
-
+	accessKey := ctx.String("access-key")
+	secretKey := ctx.String("secret-key")
+	description := ctx.String("description")
 	expDurVal := ctx.Duration("expiry-duration")
-	expVal := ctx.String("expiry")
+
+	// generate access key and secret key
+	if len(accessKey) <= 0 || len(secretKey) <= 0 {
+		randomAccessKey, randomSecretKey, err := generateCredentials()
+		if err != nil {
+			fatalIf(err, "unable to generate randomized access credentials")
+		}
+		if len(accessKey) <= 0 {
+			accessKey = randomAccessKey
+		}
+		if len(secretKey) <= 0 {
+			secretKey = randomSecretKey
+		}
+	}
 
 	if expVal != "" && expDurVal != 0 {
-		e := fmt.Errorf("Only one of --expiry or --expiry-duration can be specified")
-		fatalIf(probe.NewError(e), "Invalid flags.")
+		fatalIf(probe.NewError(errors.New("Only one of --expiry or --expiry-duration can be specified")), "invalid flags")
 	}
 
-	var exp time.Time
-	if expVal != "" {
-		location, e := time.LoadLocation("Local")
-		if e != nil {
-			fatalIf(probe.NewError(e), "Unable to parse the expiry argument.")
+	opts := madmin.AddServiceAccountReq{
+		TargetUser:  targetUser,
+		AccessKey:   accessKey,
+		SecretKey:   secretKey,
+		Name:        name,
+		Description: description,
+	}
+
+	if policyPath != "" {
+		// Validate the policy document and ensure it has at least when statement
+		policyBytes, e := os.ReadFile(policyPath)
+		fatalIf(probe.NewError(e), "unable to read the policy document")
+
+		p, e := policy.ParseConfig(bytes.NewReader(policyBytes))
+		fatalIf(probe.NewError(e), "unable to parse the policy document")
+
+		if p.IsEmpty() {
+			fatalIf(errInvalidArgument(), "empty policies are not allowed")
 		}
 
-		patternMatched := false
+		opts.Policy = policyBytes
+	}
+
+	switch {
+	case expVal != "":
+		location, e := time.LoadLocation("Local")
+		fatalIf(probe.NewError(e), "unable to load local location verify your local TZ=<val> settings")
+
+		var found bool
 		for _, format := range supportedTimeFormats {
 			t, e := time.ParseInLocation(format, expVal, location)
 			if e == nil {
-				patternMatched = true
-				exp = t
+				found = true
+				opts.Expiration = &t
 				break
 			}
 		}
 
-		if !patternMatched {
-			e := fmt.Errorf("invalid expiry date format '%s'", expVal)
-			fatalIf(probe.NewError(e), "unable to parse the expiry argument.")
+		if !found {
+			fatalIf(probe.NewError(fmt.Errorf("invalid expiry date format '%s'", expVal)), "unable to parse the expiry argument")
 		}
-	} else if expDurVal != 0 {
-		exp = time.Now().Add(expDurVal)
-	} else {
-		exp = time.Unix(0, 0)
+	case expDurVal != 0:
+		t := time.Now().Add(expDurVal)
+		opts.Expiration = &t
+	default:
+		t := time.Unix(0, 0)
+		opts.Expiration = &t
```

Comment:
I think this will still fail the opts validation in madmin if no expiry is specified. Removing the default statement and leaving opts.Expiration nil would fix the problem

Reply:
then its a bug in `madmin-go` not in `mc` @taran-p

Reply:
however let me make provisions in mc for now.